alfresco-transform-aio/alfresco-transform-aio/src/main/java/org/alfresco/transformer/transformers/AllInOneTransformer.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * #%L
+ * Alfresco Transform Core
+ * %%
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * -
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transformer.transformers;
+
+import org.alfresco.transform.client.model.config.TransformConfig;
+import org.alfresco.transform.client.model.config.TransformOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Top level transformer managing multiple sub transformers.
+ *
+ * @author eknizat
+ */
+public class AllInOneTransformer implements Transformer
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(AllInOneTransformer.class);
+
+    /**
+     * Represents the mapping between a transform and a transformer, multiple mappings can point to the same transformer.
+     */
+    private Map<String, Transformer> transformerTransformMapping = new HashMap();
+
+    public AllInOneTransformer()
+    {
+        // TODO - use observer style registration?
```

Comment:
Can we remove TODO before merge to master (else raise tech debt as a JIRA and add the id here for reference) ?